File: 2025_08/leetcode2025-08-05.py
from typing import List
from collections import defaultdict

class Solution:
    def numOfUnplacedFruits(self, fruits: List[int], baskets: List[int]) -> int:
        res = len(fruits)
        n = len(fruits)
        used = defaultdict(int)
        for i in range(n):
            for j in range(n):
                if used[j]==0 and baskets[j]>=fruits[i]:
                    used[j] = 1
                    res -= 1
                    break
        return res

if __name__ == "__main__":
    s = Solution()
    fruits = [4,2,5]
    baskets = [3,5,4]
    print(s.numOfUnplacedFruits(fruits, baskets))

File: leetcode2025/2025_08/leetcode2025-08-05.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int numOfUnplacedFruits(vector<int>& fruits, vector<int>& baskets) {
        int res = fruits.size();
        int n = fruits.size();
        unordered_map<int, int> used;
        for (int i=0; i<n; i++){
            for (int j=0; j<n; j++){
                if (used[j] == 0 && baskets[j] >= fruits[i]){
                    used[j] = 1;
                    res--;
                    break;
                }
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> fruits = {4,2,5};
    vector<int> baskets = {3,5,4};
    cout << s.numOfUnplacedFruits(fruits, baskets) << endl;
}
